Engine/CommandInfoCache.cs

```diff
@@ -12,7 +12,7 @@ namespace Microsoft.Windows.PowerShell.ScriptAnalyzer
     /// <summary>
     /// Provides threadsafe caching around CommandInfo lookups with `Get-Command -Name ...`.
     /// </summary>
-    internal class CommandInfoCache
```

Comment:
I think we should use the recommended Dispose pattern as per documenation:
https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose#implementing-the-dispose-pattern-for-a-base-class